feat: Add Playwright Test

nitya opened the pull request:
1. Config file specifies baseURL for hosted server
2. Test specification file configured for 2 simple tests (title, #items)
3. npm script updated to run Playwright test by default

nitya committed:
Initial Playwright Setup

Default sample test spec added
Config updated for single project target (chromium only)
npm test script updated to run playwright

nitya committed:
Simple test

Runs against server specified in baseURL (config)
Shows 2 simple tests running (1 per worker as configured)
Validates there are 3 items in default list

johnpapa commented on playwright.config.ts:
```diff
@@ -0,0 +1,96 @@
+  /* Configure projects for major browsers */
+  projects: [
+    {
+      name: 'chromium',
```
will this one also cover Edge as its chromium based? or do we need the branded ones for that?

nitya replied:
Default Chromium option runs against latest version of Chromium and is recommended (valid for Chrome and Edge future proofing) but if required to run against specific versions (or stock versions) then configuration can be updated
https://playwright.dev/docs/browsers#google-chrome--microsoft-edge

johnpapa commented on tests/example.spec.ts:
```diff
@@ -0,0 +1,15 @@
+  await expect(listTitle).toHaveText("My List");
+});
+
+test('Homepage has List with 3 items', async ({ page }) => {
```
can you change this to have at least 3 items? part of the demo adds one.

johnpapa commented on tests/example.spec.ts:
```diff
@@ -0,0 +1,15 @@
+import { test, expect } from '@playwright/test';
```
Can you rename the file to `home.spec.ts` of `list.spec.ts` , so it doesnt appear to be a sample?

johnpapa committed:
npm audit fix

nitya committed:
Merge branch 'johnpapa:main' into featadd-test

nitya committed:
chore: fixing minor issues

renamed test script
added webserver dependency
checked for count greater than or equal to 3

nitya committed:
Updated Fixes

Config with multiple webservers (app and api)
Use new poll feature to wait/test for count greater than

johnpapa committed:
Merge branch 'featadd-test' of github.com:nitya/cloud-coding-with-codespaces into pr/nitya/10

johnpapa committed:
fixing the scripts?

nitya committed:
Installing dependencies for browsers

johnpapa committed:
added safari browser

johnpapa committed:
added trace for images
